lib/compress.js

```diff
@@ -601,12 +601,18 @@ merge(Compressor.prototype, {
                             ret = funs.concat([ stat.transform(compressor) ]);
                             continue loop;
                         }
+
                         //---
-                        // XXX: what was the intention of this case?
+                        // function f(e){if (a(e)) return foo();if (b(e)) return bar();if (c(e)) return baz();fail(e)};
+                        // ==> function f(e){return a(e)?foo():b(e)?bar():c(e)?baz():void fail(e)}
+                        //
                         // if sequences is not enabled, this can lead to an endless loop (issue #866).
                         // however, with sequences on this helps producing slightly better output for
                         // the example code.
+                        //
+                        // Also require to at least have 2 return statements to counter the cost of the added `void `
                         if (compressor.option("sequences")
+                            && i > 0 && statements[i - 1] instanceof AST_If && statements[i - 1].body instanceof AST_Return
```

Comment:
Initially I was concerned that `statements` was being altered elsewhere in the handle_if_return loop as it was being traversed backwards and `statements[i]` might be invalid. That doesn't appear to be the case. 

If another contributor is fine with this patch, by all means proceed.